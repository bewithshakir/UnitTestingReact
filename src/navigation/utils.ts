
 export const combinePaths = (parent: string, child: string): string =>
 `${parent.replace(/\/$/, "")}/${child.replace(/^\//, "")}`;


export const buildPaths = (routes:any, parentPath: string = ""): any =>
 routes.map((route: { path: string; routes: any; }) => {
   const path = combinePaths(parentPath, route.path);

   return {
     ...route,
     path,
     ...(route.routes && { routes: buildPaths(route.routes, path) })
   };
 });


export const setupParents = (routes:any, parentRoute:any = null): any =>
 routes.map((route: any) => {
   const withParent = {
     ...route,
     ...(parentRoute && { parent: parentRoute })
   };

   return {
     ...withParent,
     ...(withParent.routes && {
       routes: setupParents(withParent.routes, withParent)
     })
   };
 });

export const flattenRoutes = (routes:[]):any =>
 routes
   .map((route: { routes: any; }) => [route.routes ? flattenRoutes(route.routes) : [], route])
   .flat(Infinity);

/**
* Combine all the above functions together
*
* @param routes
* @returns {any[]}
*/
export const generateAppRoutes = (routes: any) => {
 return flattenRoutes(setupParents(buildPaths(routes)));
};

/**
* Provides path from root to the element
*
* @param route
* @returns {any[]}
*/
export const pathTo = (route: { parent: any; }) : any=> {
 if (!route.parent) {
   return [route];
 }

 return [...pathTo(route.parent), route];
};