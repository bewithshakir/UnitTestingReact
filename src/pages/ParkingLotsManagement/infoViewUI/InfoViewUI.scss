div {
  .right_info_panel_content {
    .infoPanelUI {
      padding: 20px 0px 20px 35px;
      .MuiGrid-container {
        padding: 16px 0px;
        font-weight: bold;
        .right_info_panel_content_label {
          color: var(--Lightergray);
          word-break: break-word;
        }
        .right_info_panel_content_value {
          color: var(--Darkgray);
        }
      }
      .right_info_panel_content_item_divider {
        width: 100%;
      }
      .opis-rack-info-text {
        border: 1px solid #fbce06;
        background: #fbce061a 0% 0% no-repeat padding-box;
    }
    }
    
    
  }
}
