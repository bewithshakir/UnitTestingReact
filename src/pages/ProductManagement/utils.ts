const setHelperTextByFieldType = (formik: any, name: string, fieldType: string) => {
    if (fieldType === 'select') {
        return (formik.errors as any)[name]?.value;
    }
    if (fieldType === 'input') {
        return (formik.errors as any)[name];
    }
    if (fieldType === 'multiselect') {
        return JSON.parse(JSON.stringify(formik.errors[name])) ;
    }
    return undefined;
};

export const getFormFieldHelperText = (formik: any, name: string, fieldType: string) => {
    if ((formik.touched as any)[name] && (formik.errors as any)[name]) {
        return setHelperTextByFieldType(formik, name, fieldType);
    }
    return undefined;
};

export const getFormFieldError = (formik: any, name: string) => {
    if ((formik.touched as any)[name] && (formik.errors as any)[name]) {
        return true;
    }
    return false;
};