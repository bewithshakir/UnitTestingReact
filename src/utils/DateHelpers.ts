import moment from 'moment';

const preFormatDate = (date: string | null | Date) => {
  return typeof date === 'string' && date.length > 0 ? date.replace('Z', '') : date;
};

export const formatDateAsYYYYMMDD = (date: string | null | Date) => {
  return date ? moment(preFormatDate(date)).format('YYYY-MM-DD') : '';
};

export const formatTime = (date: string | null | Date) => {
  return date ? moment(preFormatDate(date)).format('hh:mm A') : '';
};

export const formatDateAsMMDDYYYY = (date: string | null | Date) => {
  return date ? moment(preFormatDate(date)).format('MM/DD/YYYY') : '';
};

export const formatDateMonthDayYear = (date: string | null | Date) => {
  return date ? moment(preFormatDate(date)).format('MMM DD, YYYY') : '';
};

export const formatDateTime = (date: string | null | Date) => {
  return date ? moment(date).format('MMM DD, YYYY hh:mm A') : '';
};

export const isTheDateToday = (date: string | null | Date) => {
  return date ? moment(date).format("Do") === moment().format("Do") : '';
};

export const getCurrentYear = () => {
  return new Date().getFullYear();
};