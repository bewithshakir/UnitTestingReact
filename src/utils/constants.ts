export const maxContacts = 5;
export const pageDataLimit = 15;

export const ParkingLot_SearchParam = 'parkinglotlanding';