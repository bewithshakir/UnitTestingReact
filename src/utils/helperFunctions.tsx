import { useTheme } from '../contexts/Theme/Theme.context';
import {
    YellowFuelIcon, RedFuelIcon, GreenFuelIcon, NavyBlueFuelIcon, ParrotGreenFuelIcon,
    PurpleFuelIcon, SkyBlueFuelIcon, BrownFuelIcon, OrangeFuelIcon, AquaFuelIcon
} from '../assets/icons';

type fileFormatSize = 'Bytes'|'KB'|'MB';
export const maskPhoneNumber = (phNumber: string) => {
    const maskedPh = phNumber.match(/(\d{3})(\d{3})(\d{4})/);
    return maskedPh ? "(" + maskedPh[1] + ") " + maskedPh[2] + "-" + maskedPh[3] : phNumber;
};

export const formatFileSizeUnit = (sizeInBytes: number, fileSizeUnit: fileFormatSize = 'MB') => {
    let coversionParam = null ; 
    switch(fileSizeUnit) {
        case 'Bytes' :
            coversionParam = 1;
          break;
        case 'KB':
            coversionParam = 1000;
          break;
          case 'MB':
            coversionParam = 1000*1000;
          break;
        default:
            coversionParam = 1;
      }
      return parseFloat((sizeInBytes/(coversionParam)).toFixed(2)) + ` ${fileSizeUnit}`;
};

export const getCheckBoxDisabledByPaymentType = (value: string) => {
    return {
        'WEX': false,
        'Invoice': true,
        'Internal': true,
        'Voyager': true,
        'default': true,
    }[value || 'default'];
};

export const getSeachedDataTotalCount = (data: { pages: [{ data: { pagination: { totalCount: number } } }] }, msg: string[]) => {
    const totalCount = data?.pages[0]?.data?.pagination?.totalCount || 0;
    return (`${totalCount} ${totalCount > 1 ? msg[1] : msg[0]}`);
};


export const getCurrentAppCountry = () => {
    const { themeType } = useTheme();
    return themeType;
};

export const isCurrentAppCountryUSA = () => {
    return getCurrentAppCountry() === "USA";
};


export function getProductIcon(fuelStatus: string) {
    return ({
        "Yellow": YellowFuelIcon,
        "Red": RedFuelIcon,
        "Green": GreenFuelIcon,
        "Navy Blue": NavyBlueFuelIcon,
        "Parrot Green": ParrotGreenFuelIcon,
        "Sky Blue": SkyBlueFuelIcon,
        "Purple": PurpleFuelIcon,
        "Brown": BrownFuelIcon,
        "Orange": OrangeFuelIcon,
        "Aqua": AquaFuelIcon,
    }[fuelStatus] || YellowFuelIcon);
}

export function getLegendFontColor(fuelStatus: string) {
    const { theme } = useTheme();
    return ({
        "Yellow": theme["--Darkgray"],
        "Red": theme["--White"],
        "Green": theme["--White"],
        "Navy Blue": theme["--White"],
        "Parrot Green": theme["--Darkgray"],
        "Sky Blue": theme["--White"],
        "Purple": theme["--White"],
        "Brown": theme["--White"],
        "Orange": theme["--Darkgray"],
        "Aqua": theme["--Darkgray"],
    }[fuelStatus]);
}

export function getUploadedBy() {
    return 'Dinesh Chakkravarthi';
}

export function getUploadedIn(url: string) {
    if (url.includes('addCustomer') || url.includes('viewCustomer')) {
        return 'Add Customer';
    }
    return 'Attachments';
}

export function capitalizeFirstLetter(str: string) {
    return str && str.charAt(0).toUpperCase() + str.substring(1);
}