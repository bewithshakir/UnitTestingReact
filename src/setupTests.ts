// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { setupServer } from 'msw/node';
import { handlers } from './tests/utils';
import { setLogger } from 'react-query';

Enzyme.configure({ adapter: new Adapter() });

export const serverMsw = setupServer(...handlers);

// Establish API mocking before all tests.
beforeAll(() => serverMsw.listen());
// Reset any request handlers that we may add during the tests,
// so they don't affect other tests.
afterEach(() => serverMsw.resetHandlers());
// Clean up after the tests are finished.
afterAll(() => serverMsw.close());

// Turn off network error logging. 
// When testing we want to suppress network errors being logged to the console.
// silence react-query errors
setLogger({
    log: console.log,
    warn: console.warn,
    error: () => {},
});

jest.mock("react-i18next", () => ({
    useTranslation: () => ({ 
        t: (key: any) => key,
        i18n: { changeLanguage: jest.fn() }
    }),
    Trans: ({ children }: any) => children
}));

