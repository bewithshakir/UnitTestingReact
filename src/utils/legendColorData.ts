import { filter } from 'lodash';

import { getProductIcon } from "./helperFunctions";

export const formatLegendsData = (data: any)=> {
    const filteredData = filter(data?.pages[0].data.products, {  activeInactiveInd: 'Y' });

    const updatedData = filteredData.map(item => ({
        value: item.productCd,
        label: item.ProductGroup.productGroupNm + ' - ' + item.productNm, 
        icon: getProductIcon(item.ProductIcon.productIconNm),
        isDisabled: true
    }));
    return updatedData;
};

