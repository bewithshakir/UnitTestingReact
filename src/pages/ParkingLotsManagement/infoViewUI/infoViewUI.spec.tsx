import { waitFor } from "@testing-library/react";
import { renderWithClient } from '../../../tests/utils';
import InfoViewUI from './InfoViewUI.component';
const mockedUsedNavigate = jest.fn();
import { serverMsw } from '../../../setupTests';
import { rest } from "msw";

const lotId = "d0535fcf-de9c-45ca-8a52-c2275710ae23";
const lotData = {
  "deliveryLocationId": lotId,
  "deliveryLocationNm": "Barun Demo 5",
  "lotId": "76123456",
  "customerId": "167fd7be-c20e-412a-bac0-502672a055d6",
  "customerName": "Barun Demo 5",
  "streetAddress": "77 Green Acres Rd S,  Green Acres Rd S,,   Green Acres Rd S, USA",
  "cityNm": "Valley Stream",
  "stateNm": "NY",
  "postalCd": "11581",
  "rackUpdate": "27-02-2021 7:09 PM",
  "walletStatus": "Y",
  "fuelStatus": [
    {
      "productCd": "0dd5f587-8c6f-40ce-9f77-1616f4b5b842",
      "productNm": "Barun 1",
      "productIcon": {
        "productIconCd": "cdc00914-dbef-4603-89c5-9f18e4af3ccc",
        "productIconNm": "Parrot Green",
        "productIconHexCode": "#BED50F",
        "activeInactiveInd": "Y"
      }
    },
    {
      "productCd": "8ee38e73-f9dd-40ca-a53c-49327eab1487",
      "productNm": "Regular1",
      "productIcon": {
        "productIconCd": "1bcfe6f1-0fae-4473-a686-82084c8a5030",
        "productIconNm": "Aqua",
        "productIconHexCode": "#89CFDC",
        "activeInactiveInd": "Y"
      }
    }
  ],
  "vehicles": 20,
  "primaryContactName": "Barun Sharma",
  "primaryContactNumber": "9900145505",
  "primaryContactEmail": "barunshrm@gmail.com"
};

const productsData = {
  data: {
    pagination: {},
    lotProducts: [
      {
        applicableProductId: 'c939d8dd-53f1-41f5-9fe6-a5dfd99ac812',
        deliveryLocationId: lotId,
        productId: '0dd5f587-8c6f-40ce-9f77-1616f4b5b842',
        productNm: 'Diesel',
        productColor: {
          productIconCd: 'cdc00914-dbef-4603-89c5-9f18e4af3ccc',
          productIconNm: 'Parrot Green',
          productIconHexCode: '#BED50F',
          activeInactiveInd: 'Y',
        },
        pricingModel: {
          pricingModelCd: '7f5411c8-ae32-43f6-a05e-146b4d866206',
          pricingModelNm: 'OPIS Rack',
          countryCd: 'us',
          activeInactiveIndicator: 'Y',
        },
        activeInactiveInd: 'Y',
        opisRackStatus: 'N',
        opisProductKey:
          'biodieselb0-5smeuls#2wintersx105opisaverage0no2nynewyorksme',
        totalPricePerGallon: '16.8416',
        manualPriceAmt: 4.727,
        addedPriceAmt: 1,
        discountPriceAmt: null,
      },
      {
        applicableProductId: '5eab3686-7ec9-4cdc-80d5-833dc080b54b',
        deliveryLocationId: lotId,
        productId: '8ee38e73-f9dd-40ca-a53c-49327eab1487',
        productNm: 'Regular1 Retail',
        productColor: {
          productIconCd: '1bcfe6f1-0fae-4473-a686-82084c8a5030',
          productIconNm: 'Aqua',
          productIconHexCode: '#89CFDC',
          activeInactiveInd: 'Y',
        },
        pricingModel: {
          pricingModelCd: '945d3d8b-57bb-42c8-88a5-2dae9cd230f0',
          pricingModelNm: 'OPIS Retail',
          countryCd: 'us',
          activeInactiveIndicator: 'Y',
        },
        activeInactiveInd: 'Y',
        opisRackStatus: 'Y',
        opisProductKey: null,
        totalPricePerGallon: '5.4540',
        manualPriceAmt: 4.954,
        addedPriceAmt: 2.1,
        discountPriceAmt: 1.6,
      },
    ],
  },
  error: null,
}


describe('load inforView component', () => {

  it('load component', async () => {
    const result = renderWithClient(<InfoViewUI rowLotId={lotId} lotData={lotData} />);
    expect(result).toMatchSnapshot();
  });

  it('load pricing data', async () => {
    serverMsw.use(
      rest.get('*', (req, res, ctx) => {
        return res(
          ctx.status(200),
          ctx.json(productsData)
        );
      })
    );
    const result = renderWithClient(<InfoViewUI rowLotId={lotId} lotData={lotData} />);
    await waitFor(() => {
      expect(result.getByText(/Diesel/i)).toBeInTheDocument();
      expect(result.getByText('$16.8416/gal')).toBeInTheDocument();
      expect(result.getByText(/OPIS Rack/i)).toBeInTheDocument();
    });
  });
});