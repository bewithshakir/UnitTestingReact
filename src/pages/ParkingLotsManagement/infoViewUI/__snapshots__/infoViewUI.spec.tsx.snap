// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load inforView component load component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="infoPanelUI"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 disclaimerTextBox css-17261w9-MuiTypography-root"
        >
          parkingLotManagement.infoView.addproductopistext
        </h5>
        <div
          class="MuiGrid-root MuiGrid-container mainDataDiv css-11lq3yg-MuiGrid-root"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="infoPanelUI"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 disclaimerTextBox css-17261w9-MuiTypography-root"
      >
        parkingLotManagement.infoView.addproductopistext
      </h5>
      <div
        class="MuiGrid-root MuiGrid-container mainDataDiv css-11lq3yg-MuiGrid-root"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
